packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -562,34 +626,109 @@ class RenderListWheelViewport
-      _centerOriginTransform(transform),
-      // Pre-transform painting function.
-      (PaintingContext context, Offset offset) {
-        context.paintChild(
-          child,
-          offset + new Offset(
-            untransformedPaintingCoordinates.dx,
-            // Paint everything in the center (e.g. angle = 0), then transform.
-            -_topScrollMarginExtent,
-          ),
-        );
-      },
-    );
+    void _paintCylindrically(PaintingContext context, Offset offset) {
+      context.pushTransform(
+        // Text with TransformLayers and no cullRects currently have an issue rendering
+        // https://github.com/flutter/flutter/issues/14224.
+        false,
+        offset,
+        _centerOriginTransform(transform),
+        // Pre-transform painting function.
+            (PaintingContext context, Offset offset) {
+          context.paintChild(
+            child,
+            offset + new Offset(
+              untransformedPaintingCoordinates.dx,
+              // Paint everything in the center (e.g. angle = 0), then transform.
+              -_topScrollMarginExtent,
+            ),
+          );
+        },
+      );
+    }
+
+    if (!useMagnifier) {
+      _paintCylindrically(context, offset);
+      return;
+    }
+
+    final double magnifierTopLinePosition =
+        size.height / 2 - _itemExtent * _magnifyRate / 2;
+    final double magnifierBottomLinePosition =
+        size.height / 2 + _itemExtent * _magnifyRate / 2;
+
+    // Some part of the child is in the center magnifier.
+    if (untransformedPaintingCoordinates.dy
+          >= magnifierTopLinePosition - _itemExtent * _magnifyRate
+        && untransformedPaintingCoordinates.dy <= magnifierBottomLinePosition) {
+      final Rect centerRect = new Rect.fromLTWH(
+          0.0,
+          magnifierTopLinePosition,
+          size.width,
+          _itemExtent * _magnifyRate);
+      final Rect topHalfRect = new Rect.fromLTWH(
+          0.0,
+          0.0,
+          size.width,
+          magnifierTopLinePosition);
+      final Rect bottomHalfRect = new Rect.fromLTWH(
+          0.0,
+          magnifierBottomLinePosition,
+          size.width,
+          magnifierTopLinePosition);
+
+      // Clipping the part in the center.
+      context.pushClipRect(
+          false,
+          offset,
+          centerRect,
+          (PaintingContext context, Offset offset) {
+            context.pushTransform(
+                false,
+                offset,
+                _magnifyTransform(),
+                (PaintingContext context, Offset offset) {
+                  context.paintChild(
+                      child,
+                      offset + untransformedPaintingCoordinates);
+                });
+          });
+
+      // Clipping the part in either the top-half or bottom-half of the wheel.
+      context.pushClipRect(
+          false,
+          offset,
+          untransformedPaintingCoordinates.dy <= magnifierTopLinePosition
```

Comment:
Ahh I see, making the magnify a number instead of a boolean will make the math here more complicated. Up to you then if you want to do it.